rt/rs/security/http-signature/src/main/java/org/apache/cxf/rs/security/httpsignature/MessageVerifier.java

```diff
@@ -119,11 +118,23 @@ public void verifyMessage(Map<String, List<String>> messageHeaders, String metho
 
         // Add the default required headers
         boolean requestor = MessageUtils.isRequestor(m);
+        byte[] payload = null;
         if (addDefaultRequiredHeaders) {
             if (!(requestor || signedHeaders.contains(HTTPSignatureConstants.REQUEST_TARGET))) {
                 signedHeaders.add(HTTPSignatureConstants.REQUEST_TARGET);
             }
-            if (!signedHeaders.contains("digest")) {
+
+            InputStream is = m.getContent(InputStream.class);
```

Comment:
This is not needed, in every place the `verifyMessage` is called (`VerifySignatureClientFilter` / `VerifySignatureFilter`), the message body is available, you could just add the new `byte[] payload` argument and pass it through.

Reply:
@reta Suggested change has been incorporated

Reply:
👍 Thank you!